agent-inject/agent/agent.go

```diff
@@ -355,6 +368,67 @@ func ShouldInject(pod *corev1.Pod) (bool, error) {
 	return true, nil
 }
 
+// ShouldDelete checks whether the pod in question has already been injected
+// and the generated config map needs to be deleted.
+func ShouldDelete(pod *corev1.Pod) (bool, error) {
+	raw, ok := pod.Annotations[AnnotationAgentInject]
+	if !ok {
+		return false, nil
+	}
+
+	inject, err := strconv.ParseBool(raw)
+	if err != nil {
+		return false, err
+	}
+
+	if !inject {
+		return false, nil
+	}
+
+	// This shouldn't happen so bail.
+	raw, ok = pod.Annotations[AnnotationAgentStatus]
+	if !ok {
+		return false, nil
+	}
+
+	// Previously injected agents
+	if raw != "injected" {
```

Comment:
Would it make sense to have a small "enum" type for the possible values of status? IIRC there are only like 2 states, but the definition of those states would make for a nice reference of what the possible states are.

Reply:
I think that could be a nice clean up in the future since `shouldInject` is very similar to this.